### PR TITLE
Added option for admins to manually add new users

### PR DESCRIPTION
**Pull request by marceloschmidt**

### CODE REVIEW

**Comment by marceloschmidt** [comment]
Closes #1509

File: packages/rocketchat-lib/server/methods/insertOrUpdateUser.coffee [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		canEditUser = RocketChat.authz.hasPermission( user._id, 'edit-other-user-info')
+		canAddUser = RocketChat.authz.hasPermission( user._id, 'add-user')
+
+		if userData._id and user._id isnt userData._id and canEditUser isnt true
+			throw new Meteor.Error 'not-authorized', '[methods] updateUser -> Not authorized'
+
+		if not userData._id and canAddUser isnt true
+			throw new Meteor.Error 'not-authorized', '[methods] updateUser -> Not authorized'
+
+		unless userData.name
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/insertOrUpdateUser.coffee]: This allows names with only spaces ('  ')

File: packages/rocketchat-lib/server/methods/insertOrUpdateUser.coffee [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			throw new Meteor.Error 'name-is-required', 'Name field is required'
+
+		unless userData.username
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/insertOrUpdateUser.coffee]: This allows usernames with only spaces ('  ')

**Comment by marceloschmidt** [comment]
@rodrigok I fixed where you commented and also added a new check for usernames to match `UTF8_Names_Validation` setting.

**Comment by rodrigok** [comment]
LGTM

**Comment by engelgabriel** [comment]
LGTM

**Pull request merged**